lib/Event/Model/Asset/ResolveUploadTargetEvent.php

```diff
@@ -25,20 +25,28 @@ class ResolveUploadTargetEvent extends Event
 
     protected string $filename;
 
-    protected array $context;
+    protected array $context = [];
 
     protected int $parentId;
 
     /**
      * ResolveUploadTargetEvent constructor.
      *
-     * @param array $context contextual information
+     * @param array|null $context contextual information
      */
-    public function __construct(int $parentId, string $filename, array $context)
+    public function __construct(int $parentId, string $filename, ?array $context)
```

Comment:
I think you should default to `null`, otherwise users are still forced to explicitly pass the third argument.

```suggestion
    public function __construct(int $parentId, string $filename, ?array $context = null)
```